File: app/src/main/java/pl/orbitemobile/wspolnoty/activities/contact/ContactContract.java
/*
 * Copyright (c) 2017. All Rights Reserved. Michal Jankowski orbitemobile.pl
 */

package pl.orbitemobile.wspolnoty.activities.contact;

import pl.orbitemobile.wspolnoty.BasePresenter;
import pl.orbitemobile.wspolnoty.BaseView;

public class ContactContract {
    
    interface View extends BaseView<Presenter> {
        
    }
    
    interface Presenter extends BasePresenter {
        
        void onPhoneClick();
        
        void onMailClick();
        
        void onWebsiteClick();
    }
}

File: app/src/main/java/pl/orbitemobile/wspolnoty/activities/contact/ContactContract.kt
/*
 * Copyright (c) 2017. All Rights Reserved. Michal Jankowski orbitemobile.pl
 */

package pl.orbitemobile.wspolnoty.activities.contact

import pl.orbitemobile.mvp.MvpPresenter
import pl.orbitemobile.mvp.MvpView

class ContactContract {

    abstract class View(layoutId: Int) : MvpView<Presenter>(layoutId)

    interface Presenter : MvpPresenter<View> {

        fun onPhoneClick()

        fun onMailClick()

        fun onWebsiteClick()
    }
}
